### PR TITLE
Petrycta/separate qore module loadings 2966

### PR DESCRIPTION
**Pull request by petrycta**

### CODE REVIEW

File: lib/ModuleManager.cpp [hunk 1]
```diff
@@ -839,11 +840,82 @@ void QoreModuleManager::loadModuleIntern(ExceptionSink& xsink, const char* name,
...
+            loadModuleIntern(xsink, modulePath.c_str(), pgm, reexport, op, version, src, mpgm, load_opt);
+            if (!xsink)
+                return true;
+        }
+        ++path;
+    }
+    return false;
+}
+
+bool exists(const QoreString& filePath) {
```
Comment (petrycta) [inline lib/ModuleManager.cpp]: Should it be static?

Comment (davidnich) [reply lib/ModuleManager.cpp]: yes, please make it static for now and change the name to `exists_file()` or `exists_path()` or something like that

File: lib/scanner.lpp [hunk 2]
```diff
@@ -1205,6 +1205,13 @@ RTIME           PT(-?[0-9]+(\.[0-9]+)?[HMSu])+
                                            else {
                                               // re-get the buffer pointer
                                               fn = fname->getBuffer();
+
+                                              // in case fn is in form "ModuleName.SomeModuleClass.qc"
+                                              // convert it to absolute path "/some/implicit/path/ModuleName/SomeModuleClass.qc". see #2966
+                                              QoreString moduleAbsPath(QMM.getModuleAbsPath(fn));
+                                              if (!moduleAbsPath.empty())
+                                                fn = moduleAbsPath.c_str();
```
Comment (davidnich) [inline lib/scanner.lpp:1213]: please indent with 4 spaces - some of the old code is indented with 3 spaces, in these cases you can update the indentation of the function or block if you want

**Review by davidnich — CHANGES_REQUESTED** [review]
let's find a way to include all the files in the module directory automatically - without having to manually `%include` them - one way to do it would be to read in each file separately and parse into the module's `Program` object with `Program::parsePending()` and then at the end call `Program::parseCommit()`

File: lib/ModuleManager.cpp (hunk 1 above)
Comment (petrytar) [reply lib/ModuleManager.cpp]: Is it ok to move it to QoreDir?

Comment (petrycta) [reply lib/ModuleManager.cpp]: Is it ok to move it to QoreDir?

Comment (davidnich) [reply lib/ModuleManager.cpp]: yes, sure, good idea

**Pull request closed without merging**